cpp/src/arrow/util/thread_pool_test.cc

```diff
@@ -722,6 +722,9 @@ TEST_F(TestThreadPoolForkSafety, MultipleChildThreads) {
 
 TEST_F(TestThreadPoolForkSafety, NestedChild) {
   {
+#ifndef __APPLE__
```

Comment:
Typo? `#ifndef` -> `#ifdef`?
See test failure on macos from a recent pr
https://github.com/apache/arrow/runs/5818260165?check_suite_focus=true#step:8:420

Reply:
Oops.  Thank you: opened #12801